examples/vision/adamatch.py

```diff
@@ -0,0 +1,453 @@
+
+import tensorflow_datasets as tfds
+
+tfds.disable_progress_bar()
+
+"""
+## Datasets
+"""
+
+(
+    (mnist_x_train, mnist_y_train),
+    (mnist_x_test, mnist_y_test),
+) = keras.datasets.mnist.load_data()
+
+# Add a channel dimension
+mnist_x_train = tf.expand_dims(mnist_x_train, -1)
+mnist_x_test = tf.expand_dims(mnist_x_test, -1)
+
+# Convert the labels to one-hot encoded vectors
+mnist_y_train = tf.one_hot(mnist_y_train, 10).numpy()
+
+svhn_train, svhn_test = tfds.load(
+    "svhn_cropped", split=["train", "test"], as_supervised=True
+)
+
+"""
+## Constants and hyperparameters
+"""
+
+TF_PI = tf.constant(np.pi)
+RESIZE_TO = 32
+
+SOURCE_BATCH_SIZE = 64
+TARGET_BATCH_SIZE = 3 * SOURCE_BATCH_SIZE  # Reference: Section 3.2
+EPOCHS = 10
+STEPS_PER_EPOCH = len(mnist_x_train) // SOURCE_BATCH_SIZE
+TOTAL_STEPS = EPOCHS * STEPS_PER_EPOCH
+T = 0
+
+AUTO = tf.data.AUTOTUNE
+LEARNING_RATE = 0.03
+
+"""
+## Data augmentation utilities
+
+A standard element of SSL algorithms is to feed weakly and strongly augmented versions of
+the same images to the learning model and making its predictions consistent. For strong
+augmentation, [RandAugment](https://arxiv.org/abs/1909.13719) is a standard choice. For
+weak augmentation, we will use horizontal flipping and random cropping.
+"""
+
+# Initialize `RandAugment` object with 2 layers of
+# augmentation transforms and strength of 5.
+augmenter = RandAugment(num_layers=2, magnitude=5)
+
+
+def weak_augment(image, source=True):
+    if image.dtype != tf.float32:
+        image = tf.cast(image, tf.float32)
+
+    # MNIST images are grayscale, this is why we first convert them to
+    # RGB images.
+    if source:
+        image = tf.image.resize_with_pad(image, RESIZE_TO, RESIZE_TO)
+        image = tf.tile(image, [1, 1, 3])
+    image = tf.image.random_flip_left_right(image)
+    image = tf.image.random_crop(image, (RESIZE_TO, RESIZE_TO, 3))
+    return image
+
+
+def strong_augment(image, source=True):
+    if image.dtype != tf.float32:
+        image = tf.cast(image, tf.float32)
+
+    if source:
+        image = tf.image.resize_with_pad(image, RESIZE_TO, RESIZE_TO)
+        image = tf.tile(image, [1, 1, 3])
+    image = augmenter.distort(image)
+    return image
+
+
+"""
+## Data loader utilities
+"""
+
+
+def create_individual_ds(ds, aug_func, source=True):
+    if source:
+        batch_size = SOURCE_BATCH_SIZE
+    else:
+        # During training 3x more target unlabeled samples are shown
+        # to the model in AdaMatch (Section 3.2 of the paper).
+        batch_size = TARGET_BATCH_SIZE
+    ds = ds.shuffle(batch_size * 10, seed=42)
+
+    if source:
+        ds = ds.map(lambda x, y: (aug_func(x), y), num_parallel_calls=AUTO)
+    else:
+        ds = ds.map(lambda x, y: (aug_func(x, False), y), num_parallel_calls=AUTO)
+
+    ds = ds.batch(batch_size).prefetch(AUTO)
+    return ds
+
+
+"""
+`_w` and `_s` suffixes denote weak and strong respectively.
```

Comment:
Prefer using fully spelled-out variable names for better readability.

Reply:
It was creating very big variable names and was breaking the readability in some cases. This is why I used shorthand notations.